libyul/AssemblyStack.h

```diff
@@ -91,15 +96,16 @@ class AssemblyStack
 
 	/// Run the assembly step (should only be called after parseAndAnalyze).
 	/// In addition to the value returned by @a assemble, returns
-	/// a second object that is guessed to be the runtime code.
+	/// a second object that is the runtime code.
 	/// Only available for EVM.
-	std::pair<MachineAssemblyObject, MachineAssemblyObject> assembleAndGuessRuntime() const;
+	std::pair<MachineAssemblyObject, MachineAssemblyObject> assembleWithDeployed(std::optional<std::string_view> _deployName = {}) const;
 
 	/// Run the assembly step (should only be called after parseAndAnalyze).
 	/// In addition to the value returned by @a assemble, returns
 	/// a second object that is the runtime code.
 	/// Only available for EVM.
-	std::pair<MachineAssemblyObject, MachineAssemblyObject> assembleWithDeployed(std::optional<std::string_view> _deployeName = {}) const;
+	std::pair<std::shared_ptr<evmasm::Assembly>, std::shared_ptr<evmasm::Assembly>>
+		assembleEVM(std::optional<std::string_view> _deployName = {}) const;
```

Comment:
Having three very similarly named functions for assembling stuff (`assemble()`, `assembleEVM()` and `assembleWithDeployed()`) is confusing. I was hard for me to tell what the actual differences were until I read their code. For example `assembleEVM()` sounds like it would be a version of `assemble()` specific to EVM but turns out that it's basically a version of `assembleWithDeployed()` that stops earlier.

What do you think about renaming:
- `assembleWithDeployed()` to `assembleWithDeployedAndGenerateBytecode()`. It would be awfully long but at at least descriptive and would make the relationship between the functions clear.
- `assembleEVM()` to `assembleWithDeployed()`.

I think their docstrings should also be updated to mention the difference. Currently the docstrings for `assembleWithDeployed()` and `assembleEVM()` are identical.

Reply:
I think `...GenerateBytecode` is confusing. What about `assemblyToMachineObject`?

Reply:
Fine with me. I actually had trouble with describing the difference accurately and the biggest difference that came to mind was that it assigns the `bytecode` field.

But I think both functions should have the `WithDeployed` part or both should drop it.